test/torchaudio_unittest/prototype/io_test.py

```diff
@@ -0,0 +1,348 @@
+        s.add_basic_video_stream(frames_per_chunk=-1, width=3, height=5)
+        s.add_basic_video_stream(frames_per_chunk=-1, frame_rate=7)
+        s.add_basic_video_stream(frames_per_chunk=-1, format="BGR")
+
+        sinfo = s.get_out_stream_info(0)
+        assert sinfo.source_index == s.default_video_stream
+        assert sinfo.filter_description == ""
+
+        sinfo = s.get_out_stream_info(1)
+        assert sinfo.source_index == s.default_video_stream
+        assert "scale=width=3:height=5" in sinfo.filter_description
+
+        sinfo = s.get_out_stream_info(2)
+        assert sinfo.source_index == s.default_video_stream
+        assert "fps=7" in sinfo.filter_description
+
+        sinfo = s.get_out_stream_info(3)
+        assert sinfo.source_index == s.default_video_stream
+        assert "format=pix_fmts=bgr24" in sinfo.filter_description
+
+    def test_remove_streams(self):
+        """`remove_stream` removes the correct output stream"""
+        s = Streamer(get_video_asset())
+        s.add_basic_audio_stream(frames_per_chunk=-1, sample_rate=24000)
+        s.add_basic_video_stream(frames_per_chunk=-1, width=16, height=16)
+        s.add_basic_audio_stream(frames_per_chunk=-1, sample_rate=8000)
+
+        sinfo = [s.get_out_stream_info(i) for i in range(3)]
+        s.remove_stream(1)
+        del sinfo[1]
+        assert sinfo == [s.get_out_stream_info(i) for i in range(s.num_out_streams)]
+
+        s.remove_stream(1)
+        del sinfo[1]
+        assert sinfo == [s.get_out_stream_info(i) for i in range(s.num_out_streams)]
+
+        s.remove_stream(0)
+        del sinfo[0]
+        assert [] == [s.get_out_stream_info(i) for i in range(s.num_out_streams)]
+
+    def test_remove_stream_invalid(self):
+        """Attempt to remove invalid output streams raises IndexError"""
+        s = Streamer(get_video_asset())
+        for i in range(-3, 3):
+            with self.assertRaises(IndexError):
+                s.remove_stream(i)
+
+        s.add_audio_stream(frames_per_chunk=-1)
+        for i in range(-3, 3):
+            if i == 0:
+                continue
+            with self.assertRaises(IndexError):
+                s.remove_stream(i)
+
+
+@skipIfNoFFmpeg
+class StreamerAudioTest(TempDirMixin, TorchaudioTestCase):
+    """Test suite for audio streaming"""
+
+    def _get_reference_wav(self, sample_rate, channels_first=False, **kwargs):
+        data = get_wav_data(**kwargs, normalize=False, channels_first=channels_first)
+        path = self.get_temp_path("ref.wav")
+        save_wav(path, data, sample_rate, channels_first=channels_first)
+        return path, data
+
+    def _test_wav(self, path, original, dtype):
+        s = Streamer(path)
+        s.add_basic_audio_stream(frames_per_chunk=-1, dtype=dtype)
+        s.process_all_packets()
+        (output,) = s.pop_chunks()
+        self.assertEqual(original, output)
+
+    @nested_params(
+        ["int16", "uint8", "int32"],  # "float", "double", "int64"]
```

Comment:
why are the additional types commented out?

Reply:
The code supports these, but test fixture (save_wav which is backed by scipy) does not support it so we cannot test.